daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -994,6 +994,6 @@ final class VariableRuntimeData(
       }
     }
 
-  def newVariableInstance: VariableInstance = VariableInstance(state, value, this, maybeDefaultValueExpr)
+  def newVariableInstance: VariableInstance = VariableInstance(state, value, this, maybeDefaultValueExpr, state, value)
```

Comment:
This isn't clear to me. Why is state and value passed in as both the current state/value and the prior state/value? Seems to me when we create a new variable instance we want the prior state/value to be undefined/no value.

Reply:
So this "newVariableInstance" is confusingly what creates the initial VariableInstace. It isn't really related to the feature NVI outside of being called to create a VariableInstance object. It is called whenever a variable is created. I think renaming this to createVariableInstance may relieve the confusion. As for the prior state/value I think that may have been a change I left in while debugging some of the issues. I am removing it now.